NodeBase/start-vnc.sh

```diff
@@ -44,6 +44,17 @@ if [ "${START_XVFB:-$SE_START_XVFB}" = true ] ; then
       echo "Waiting for Xvfb..."
     done
 
+    # Guard against unreasonably high nofile limits. See https://github.com/SeleniumHQ/docker-selenium/issues/2045
+    if [[ $(ulimit -n) -gt 200000 ]]; then
+      echo "Trying to lower open file descriptor limit from $(ulimit -n) to 65536."
+      ulimit -n 65536
+      if [ $? -eq 0 ]; then
+        echo "Successfully lowered the open descriptor limit."
+      else
+        echo "Failed to lower the open file descriptor limit. This can result in delays when connecting to VNC."
+      fi
+    fi
+
```

Comment:
What happens if the command fails? Does execution break or it just resumes and echoes the 2nd message?

Reply:
It just carries on, this file doesn't have stop on error enabled. A few lines above is a similar construct.

I'm not even 100% sure that this command can even fail. Lowering the limit seems like something that always works but I'm not knowledgeable enough to say for certain so I added a bit of logging.

Reply:
I have tried a negative test by update wrong command, 2nd message echoes and next step can go
```bash
Trying to lower open file descriptor limit from 1048576 to 65536.
/opt/bin/start-vnc.sh: line 50: ulimit: -1: invalid option
ulimit: usage: ulimit [-SHabcdefiklmnpqrstuvxPT] [limit]
Failed to lower the open file descriptor limit. This can result in delays when connecting to VNC.
11/12/2023 16:56:48 passing arg to libvncserver: -rfbport
11/12/2023 16:56:48 passing arg to libvncserver: 5900
11/12/2023 16:56:48 passing arg to libvncserver: -rfbportv6
11/12/2023 16:56:48 passing arg to libvncserver: 5900
11/12/2023 16:56:48 -usepw: found /home/seluser/.vnc/passwd

```

Reply:
The file descriptor limit has been found to be high and lowering it failed, for whatever reason. That's exactly what the message is warning you about, if say your limit is set to 1 billion for one reason or another. Continuing on the user should be aware that things might not work like they expect them to

Reply:
My test only verify status != 0 not break the code. To avoid misleading, I think we can change another message without word `Failed` something like
`echo "The open file descriptor limit could not be updated. This can result in delays when connecting to VNC."`

Reply:
I think the limit value, we can get from ENV or default, it helps user is able to set another value e.g `-e SE_VNC_ULIMIT=100000`
something like `ulimit -n ${SE_VNC_ULIMIT:-65536}`

Reply:
Sure, done. Your version sounds better anyways

Reply:
> I think the limit value, we can get from ENV or default, it helps user is able to set another value e.g `-e SE_VNC_ULIMIT=100000` something like `ulimit -n ${SE_VNC_ULIMIT:-65536}`



> I think the limit value, we can get from ENV or default, it helps user is able to set another value e.g `-e SE_VNC_ULIMIT=100000` something like `ulimit -n ${SE_VNC_ULIMIT:-65536}`

I pushed a commit, check it out. I dropped the notice about VNC connection delays since that isn't always applicable anymore.